DependencyInjection/HttplugExtension.php

```diff
@@ -212,46 +198,34 @@ private function configureAuthentication(ContainerBuilder $container, array $con
      * @param ContainerBuilder $container
      * @param string           $name
      * @param array            $arguments
-     * @param bool             $enableCollector
+     *
+     * @return string The service id of the client.
      */
-    private function configureClient(ContainerBuilder $container, $name, array $arguments, $enableCollector)
+    private function configureClient(ContainerBuilder $container, $name, array $arguments)
     {
         $serviceId = 'httplug.client.'.$name;
-        $def = $container->register($serviceId, DummyClient::class);
-
-        // If there are no plugins nor should we use the data collector
-        if (empty($arguments['plugins']) && !$enableCollector) {
-            $def->setFactory([new Reference($arguments['factory']), 'createClient'])
-                ->addArgument($arguments['config']);
-        } else {
-            $def
-                ->setFactory('Http\HttplugBundle\ClientFactory\PluginClientFactory::createPluginClient')
-                ->addArgument(
-                    array_map(
-                        function ($id) {
-                            return new Reference($id);
-                        },
-                        $arguments['plugins']
-                    )
+        $definition = $container->register($serviceId, DummyClient::class);
+        $definition->setFactory([PluginClientFactory::class, 'createPluginClient'])
+            ->addArgument(
+                array_map(
+                    function ($id) {
+                        return new Reference($id);
+                    },
+                    $arguments['plugins']
                 )
-                ->addArgument(new Reference($arguments['factory']))
-                ->addArgument($arguments['config'])
-            ;
-
-            if ($enableCollector) {
-                $serviceIdDebugPlugin = $this->registerDebugPlugin($container, $name);
-                $def->addArgument(['debug_plugins' => [new Reference($serviceIdDebugPlugin)]]);
+            )
+            ->addArgument(new Reference($arguments['factory']))
+            ->addArgument($arguments['config'])
+            ->addArgument([])
+        ;
 
-                // tell the plugin journal what plugins we used
-                $container->getDefinition('httplug.collector.plugin_journal')
-                    ->addMethodCall('setPlugins', [$name, $arguments['plugins']]);
-            }
-        }
+        // Tell the plugin journal what plugins we used
+        $container->getDefinition('httplug.collector.plugin_journal')
```

Comment:
IMO this is still toolbar dependent. Service only declared in data_collector.xml?